File: day1/day1_solution_part1.go
package main

import (
	"encoding/csv"
	"fmt"
	"os"
	"strconv"
)

var sampleInput = []int{1, -2, 3, 1,}

func retrieveFrequencies(filepath string) [][]string {
	var (
		file, _   = os.Open(filepath)
		csvreader = csv.NewReader(file)
	)

	contents, _ := csvreader.ReadAll()

	return contents
}

func sumTotalFrequencyChanges(frequencies [][]string) {
	var sum int

	for _, f := range frequencies {
		fc, _ := strconv.Atoi(f[0])

		sum += fc
	}

	fmt.Println(sum)
}

func main() {
	filepath := os.Getenv("AOC_INPUT_FILE")
	fmt.Println(filepath)

	sumTotalFrequencyChanges(retrieveFrequencies(filepath))
}

File: day1/day1_solution_part1.py
import csv
import os 

def retrieve_frequencies(filepath):
    with open(filepath, 'r') as frequencies:
        reader = csv.reader(frequencies)
        file_contents = list(reader)

    freq_list = []
    for frequency in file_contents:
       freq_list.append(int(frequency[0]))

    return freq_list

def sum_total_frequency_changes(frequencies):
    print(sum(frequencies))

if __name__ == "__main__":
    filepath = os.getenv("AOC_INPUT_FILE")

    sum_total_frequency_changes(retrieve_frequencies(filepath))

